src/txreconciliation.cpp

```diff
@@ -69,10 +100,56 @@ class TxReconciliationTracker::Impl {
         return std::make_tuple(we_initiate_recon, we_respond_recon, RECON_VERSION, m_local_recon_salt);
     }
 
+    bool RegisterPeer(NodeId peer_id, bool peer_inbound,
+        bool they_may_initiate, bool they_may_respond, uint32_t recon_version, uint64_t remote_salt)
+    {
+        // We do not support reconciliation salt/version updates. We treat an attempt to update
+        // after a successful registration as a protocol violation.
+        LOCK(m_mutex);
+        if (m_states.find(peer_id) != m_states.end()) return false;
+
+        // If the peer supports the version which is lower than our, we downgrade to the version
+        // they support. For now, this only guarantees that nodes with future reconciliation
+        // versions have the choice of reconciling with this current version. However, they also
+        // have the choice to refuse supporting reconciliations if the common version is not
+        // satisfactory (e.g. too low).
+        recon_version = std::min(recon_version, RECON_VERSION);
+        // v1 is the lowest version, so suggesting something below must be a protocol violation.
```

Comment:
Note, the BIP330 says version != 1 should be ignored, making the 0 case a protocol violation should be reflected in the BIP, I think.

Reply:
Done in 38e66d31f7eb47cd2566003e8d0d8808b129eabb

Reply:
https://github.com/naumenkogs/bips/commit/38e66d31f7eb47cd2566003e8d0d8808b129eabb